mlir/include/mlir/Dialect/NVGPU/IR/NVGPUDialect.h

```diff
@@ -25,6 +25,8 @@ constexpr int kWarpSize = 32;
 
 /// M size of wgmma.mma_async instruction
 constexpr int kWgmmaSizeM = 64;
+/// Maximum tensor dimension that TMA supports
+constexpr int kMaxTMATensorDimension = 5;
```

Comment:
Where does this limitation come from?
Just curious, we don't necessarily need to lift it.

Reply:
tma supports max 5dim
[see the instruction](https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#data-movement-and-conversion-instructions-cp-async-bulk-tensor)